File: JavaScript/camelCase.js
/*
|--------------------------------------------------------------------------
| CamelCase Method
|--------------------------------------------------------------------------
|
*/

String.prototype.camelCase = function () {
   let res = []
   const x = this.toLowerCase().split(' ')
   for (let i = 0; i < x.length; i++) {
      if (x[i] != '') {
         res.push(`${x[i][0].toUpperCase()}${x[i].slice(1, x[i].length)}`)
      }
   }
   return res.join('')
}


String.prototype.camelCase = function () {
   return this.split(' ').map(function (word) {
      return word.charAt(0).toUpperCase() + word.slice(1);
   }).join('');
}


String.prototype.camelCase = function () {
   return this.trim().replace(/(?:^|\s+)(\w)/g, (_, c) => c.toUpperCase())
}


String.prototype.camelCase = function () {
   return this.split(/\s/)
      .map(str => str.charAt(0).toUpperCase() + str.slice(1))
      .join('');
}


console.log("test case".camelCase(), "TestCase");
console.log("camel case method".camelCase(), "CamelCaseMethod");
console.log("say hello ".camelCase(), "SayHello");
console.log(" camel case word".camelCase(), "CamelCaseWord");
console.log("".camelCase(), "");

File: Python/camel_case.py
"""
|--------------------------------------------------------------------------
| CamelCase Method
|--------------------------------------------------------------------------
|
"""

def camel_case(string):
   res = []
   x = string.split(' ')
   for i in range(len(x)):
      if x[i] != '':
         res.append(f'{x[i][0].upper()}{x[i][slice(1, len(x[i]))]}')
   return ''.join(res)


camel_case = lambda _: ''.join(map(str.title,_.split())) 


def camel_case(string):
   return string.title().replace(" ", "")


def camel_case(string):
   return ''.join([i.capitalize() for i in string.split()])


print(camel_case("test case"), "TestCase")
print(camel_case("camel case method"), "CamelCaseMethod")
print(camel_case("say hello "), "SayHello")
print(camel_case(" camel case word"), "CamelCaseWord")
print(camel_case(""), "")